[TOOLS-4365] Modifiy CMT for support cubrid 11.2

Kang-dot opened the pull request:
http://jira.cubrid.org/browse/TOOLS-4365

- Purpose
Modify CMT to support CUBRID 11.2 multi schema function.

- Implementation
This PR modify cubrid to cubrid, cubrid to dump, oracle to cubrid, oracle to dump
to support user schema function.
mssql and mysql are not applied this time.

- Remarks
N/A

Kang-dot committed:
[TOOLS-4365] Add SchemaMappingPage and CUBRID version check utility

- Add CUBRIDVersionUtils to check source, target CUBRID version and
  check it has multi schema
- Add SchemaMappingPage to mapping schema between source DB and target DB

Kang-dot committed:
[TOOLS-4365] add function to migrate db objects to selected schema

- add pk, fk, index migration function
- add view migration function
- edit SchemaMappingPage.java to using combo box component

Kang-dot committed:
[TOOLS-4365] Remove unused method in SchemaMappingPage

- Remove unused method

Kang-dot committed:
[TOOLS-4365] Add user schema in CUBRID dump file

- Add checkbox to select whether to add user schema in CUBRID dump file

Kang-dot committed:
[TOOLS-4365] Fix several migration error

- fix update statistics query
- remove develop comment
- fix serial, view migration process
- fix drop querys
- fix foreign key query
- add cubrid schema prefix
- add messages

and etc.

kisoo-han reviewed (approved):
Thank you for your efforts!!

Kang-dot committed:
[TOOLS-4365] fix several feature

- fix missing message
- fix schema mapping page regular expression
- fix null pointer exception cause by sequence

Kang-dot committed:
[TOOLS-4365] fix script migration and remove create new schema function

- fix script migration function
- remove schema creation function witch entered by the user

Kang-dot committed:
[TOOLS-4365] modify SchemaMappingPage

- fix schema mapping page
  * add grant status column
  * set target schema column's default message
  * remove create user define schema
- add messages

Kang-dot committed:
[TOOLS-4365] Merge branch 'develop' into TOOLS-4365

- merge with TOOLS-4376 corrections

Kang-dot committed:
[TOOLS-4365] fix script migration function

- fix script migration to work in CUBRID 11.2, SchemaMappingPage
- fix query which get serial from source CUBRID
- add messages

Kang-dot committed:
Merge branch 'develop' of https://github.com/CUBRID/cubrid-migration into TOOLS-4365

Kang-dot committed:
[TOOLS-4365] fix offline script migration function

- fix offline script migration function
- add messages

Kang-dot committed:
Merge branch 'develop' of https://github.com/CUBRID/cubrid-migration into TOOLS-4365

Kang-dot committed:
[TOOLS-4365] Fix FK migration function

- Fix multi schema FK export
- Fix FK migration when target cubrid is single schema setting

Kang-dot committed:
[TOOLS-4365] fix duplicate object in multi schema scenario

- Fix errors that occur when each schema has an object with the same name

Kang-dot committed:
[TOOLS-4365] fix schema mapping

- change SchemaMappingPage ui message
- fix target schema NullPointerException when progress script migration
- fix duplicate target schema name in SchemaMappingPage

Kang-dot committed:
[TOOLS-4365] fix view, serial user schema didn't append correct

- fix view user schema append correctly
- fix serial user schema append correctly
- Modify NullPointerException that occurs in schema mapping page when migration in mssql script

Kang-dot committed:
[TOOLS-4365] fix view and serial were not accurately migrate

- fix view, serial not migrate accurately

Kang-dot committed:
[TOOLS-4365] fix script migration error

- Fixed an issue that was not properly migrate if there were objects with duplicate names when migrate in offline script case

Kang-dot committed:
[TOOLS-4365] fix view DDL was not visible

- Fix an issue where view DDL was not visible on the DDL view page

Kang-dot committed:
[TOOLS-4365] edit buildTableCfg method

- check null state

Kang-dot committed:
[TOOLS-4365] fix several feature

- Minimize reliance on CUBRIDversionutils class
- fix schema mapping page
- add schema mapping information in script file

Kang-dot committed:
[TOOLS-4365] add old script warning dialog

- add old script warning dialog
  - displays a warning window if the user uses the old script
- remove some comment line

Kang-dot committed:
[TOOLS-4365] remove CUBRIDVersionUtils class

- remove CUBRIDVersionUtils class

Kang-dot committed:
[TOOLS-4365] remove comment

- remove comment about CMT 11.1 work

Kang-dot committed:
[TOOLS-4365] update verion

- update CMT version to 11.1.0.0001

Kang-dot committed:
[TOOLS-4365] fix when target schema is null in SchemaMappingPage

- Fix that the target schema is not displayed in SchemaMappingPage

Kang-dot committed:
[TOOLS-4365] rename method

- isOldScript is more currect then isOldSchema

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/core/engine/importer/impl/OfflineImporter.java:
```diff
@@ -622,6 +630,13 @@ public int importRecords(final SourceTableConfig stc, final List<Record> records
+	//CMT112 file importer schema;
+	//maybe this method didn't need?
+	public void createSchema(Schema dummySchema){
+		
```
What is this blank method?

Kang-dot replied:
I add createSchema method in `IMigrationImporter` class. it use to create user when migration target is online.(there's some case which create user in online target DB) but in offline case, "create user" query didn't need. so i just declare that method for prevent create "create user" query. see here for relevant information: `MigrationTasksScheduler.createSchema`

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/core/dbobject/Schema.java:
```diff
@@ -324,4 +366,17 @@ public Trigger getTrigger(String name) {
+
+
+	@Override
+	public String getDDL() {
```
It appears to be a method that functions the same as the "getCreateDDL" method.
Is there a reason why added it?

Kang-dot replied:
`getDDL` method is abstract method. so it must declare in `Schema` class. but i think `getCreateDDL` isn't necessary. i'm gonna find a way to combine these two methods. thanks

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/cubrid/meta/CUBRIDSchemaFetcher.java:
```diff
@@ -86,10 +87,16 @@
-
+	
 	private static final String GET_ALLSERIALINFO_SQL = "select name, owner.name, current_val, "
 			+ "increment_val, max_val,min_val, cyclic, started, class_name, att_name, cached_num, comment "
```
There are no comments in CUBRID 9 or earlier, this SQL is expected to generate an error.

Kang-dot replied:
I know. It will be resolved at the next commit

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/cubrid/meta/CUBRIDSchemaFetcher.java:
```diff
@@ -86,10 +87,16 @@
+	
+	private static final String GET_OWNER_SERIAL_SQL = "select name, owner.name, current_val, "
+			+ "increment_val, max_val, min_val, "
+			+ "cyclic, started, class_name, att_name, cached_num, comment " 
```
This SQL also seems to need a comment modification.

Kang-dot replied:
I know. It will be resolved at the next commit

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/cubrid/meta/CUBRIDSchemaFetcher.java:
```diff
@@ -103,6 +110,14 @@ public final class CUBRIDSchemaFetcher extends
+	
+	private static final String ALL_TABLE_COLUMN_WITH_USERNAME = "SELECT a.class_name, c.owner_name, a.attr_name, a.attr_type, a.from_class_name,"
+			+ " a.data_type, a.prec, a.scale, a.is_nullable,"
+			+ " a.domain_class_name, a.default_value, a.def_order,c.is_reuse_oid_class, c.comment"
```
This SQL also seems to need a comment modification.

Kang-dot replied:
I know. It will be resolved at the next commit

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/cubrid/meta/CUBRIDSchemaFetcher.java:
```diff
@@ -548,6 +569,478 @@ private void buildCUBRIDTableSerials(Connection conn, Map<String, Table> tables)
+			
+			while (rs.next()) {
+				String tableName = rs.getString("class_name");
+				String comment = rs.getString("comment");
```
Need to change to import only when CUBRID version 10 or higher.

Kang-dot replied:
I know. It will be resolved at the next commit

Kang-dot commented:
I will add a commit after resolve conflict and proceed the test.

Kang-dot committed:
[TOOLS-4365] resolve conflict

- resolve conflict

Kang-dot committed:
Merge branch 'kang-dot-TOOLS-4365' into TOOLS-4365

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/core/engine/importer/impl/OfflineImporter.java:
```diff
@@ -704,5 +712,11 @@ public void createSequence(Sequence sq) {
+		System.out.println("123456");
+		
+		// TODO Auto-generated method stub
+		// if migrate target is offline, "create user" query isn't need
```
It need to remove println method.

Kang-dot replied:
that code is removed

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/core/engine/config/MigrationConfiguration.java:
```diff
@@ -683,7 +718,8 @@ private void buildTableCfg(boolean isReset) {
 			if (StringUtils.isNotBlank(sourceDBSchema.getName())) {
-				prefix = sourceDBSchema.getName() + ".";
+//				prefix = sourceDBSchema.getName() + ".";
+				prefix = sourceDBSchema.getTargetSchemaName();
```
I think the code before modification would be clean to remove than to leave it as an annotation.

Kang-dot replied:
that code is removed

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/core/engine/report/MigrationReport.java:
```diff
@@ -259,16 +269,44 @@ public List<DBObjMigrationResult> getDbObjectsResult() {
 	}
+	
+	boolean dbObjectNullCheck(DBObjMigrationResult or, DBObject obj) {
+		if (or.getObjOwner() == null) {
```
Method description and access modifier missing.

Kang-dot replied:
Add description and rename method. it may be more intuitive now

Kang-dot committed:
[TOOLS-4365] remove stdout option in log4j configuration file

- remove stdout option in lo4j.properties

Kang-dot committed:
[TOOLS-4365] remove println method

- remove println method

Kang-dot committed:
[TOOLS-4365] remove unused import line

- remove unused import line

Kang-dot committed:
[TOOLS-4365] Modify code intuitively

- add dot after target schema name

Kang-dot committed:
[TOOLS-4365] Add method description

- Add description in new method

Kang-dot committed:
[TOOLS-4365] Add license description

- Add license description in new files

Kang-dot committed:
[TOOLS-4365] Modify method name and content more intuitively

- The method name and results have not changed, but have been modified a little more intuitively.

Kang-dot committed:
[TOOLS-4365] Fixing errors that fail to catch default messages on schema
mapping page

- add if statement in SchemaMappingPage
- remove korean message

Kang-dot committed:
[TOOLS-4365] Fixes issues where schema information is not visible on the
results page

- add setter in schema.java
- add schema in OVERVIEW_TYPES

Kang-dot committed:
[TOOLS-4365] Fixed a bug where columns could not migrate correctly

Fixed an issue where the column could not be imported properly if a
table with the same name existed in each schema when source db was
multi schema

Kang-dot committed:
[TOOLS-4365] Update version

- Update version to "11.1.0.0097"

Kang-dot committed:
Merge branch 'Kang-dot-TOOLS-4365' into TOOLS-4365
